robottelo/hosts.py

```diff
@@ -1149,47 +1149,59 @@ def install(self, installer_obj=None, cmd_args=None, cmd_kwargs=None):
 
     def capsule_setup(self, sat_host=None, **installer_kwargs):
         """Prepare the host and run the capsule installer"""
-        self.satellite = sat_host or Satellite()
-        self.create_custom_repos(
-            capsule=settings.repos.capsule_repo,
-            rhscl=settings.repos.rhscl_repo,
-            ansible=settings.repos.ansible_repo,
-            maint=settings.repos.satmaintenance_repo,
+        self._satellite = sat_host or Satellite()
+
+        # Register capsule host to CDN and enable repos
+        result = self.register_contenthost(
+            org=None,
+            lce=None,
+            username=settings.subscription.rhn_username,
+            password=settings.subscription.rhn_password,
+            auto_attach=True,
```

Comment:
Would suggest using the `register_to_cdn` method here to register to CDN (currently its only available for 6.10.z, waiting for #9709 to merge in master)
```suggestion
        result = self.register_to_cdn()
```